### PR TITLE
Support for configuration of basic authorization

### PR DESCRIPTION
**Pull request by platan**
Resolves https://github.com/nginxinc/ansible-role-nginx/issues/98
This PR gives an ability to configure basic authorization using dedicated configuration options.

### CODE REVIEW

File: templates/http/default.conf.j2 [hunk 1]
```diff
@@ -33,6 +33,12 @@ server {
 {% if item.value.reverse_proxy is defined %}
 {% for location in item.value.reverse_proxy.locations %}
     location {{ item.value.reverse_proxy.locations[location].location }} {
+{% if item.value.reverse_proxy.locations[location].auth_basic is defined %}
```
Comment (alessfg) [inline templates/http/default.conf.j2]:
```suggestion
{% if item.value.reverse_proxy.locations[location].auth_basic is defined and  item.value.reverse_proxy.locations[location].auth_basic %}
```

File: templates/http/default.conf.j2 [hunk 2]
```diff
@@ -33,6 +33,12 @@ server {
...
+        auth_basic "{{ item.value.reverse_proxy.locations[location].auth_basic }}";
+{% endif %}
+{% if item.value.reverse_proxy.locations[location].auth_basic_file is defined %}
```
Comment (alessfg) [inline templates/http/default.conf.j2]:
```suggestion
{% if item.value.reverse_proxy.locations[location].auth_basic_file is defined and item.value.reverse_proxy.locations[location].auth_basic_file %}
```

File: templates/http/default.conf.j2 [hunk 3]
```diff
@@ -56,6 +62,12 @@ server {
         index  {{ item.value.web_server.locations[location].html_file_name }};
 {% if item.value.web_server.locations[location].autoindex %}
         autoindex on;
+{% endif %}
+{% if item.value.web_server.locations[location].auth_basic is defined %}
```
Comment (alessfg) [inline templates/http/default.conf.j2]:
```suggestion
{% if item.value.web_server.locations[location].auth_basic is defined and item.value.web_server.locations[location].auth_basic %}
```

File: templates/http/default.conf.j2 [hunk 4]
```diff
@@ -56,6 +62,12 @@ server {
...
+        auth_basic "{{ item.value.web_server.locations[location].auth_basic }}";
+{% endif %}
+{% if item.value.web_server.locations[location].auth_basic_file is defined %}
```
Comment (alessfg) [inline templates/http/default.conf.j2]:
```suggestion
{% if item.value.web_server.locations[location].auth_basic_file is defined item.value.web_server.locations[location].auth_basic_file %}
```

**Comment by alessfg** [comment]
I've added a couple suggestions to the templates to cover cases where the `auth` parameters are defined but are still null. PR should be ready to merge once the suggestions are addressed.

**Comment by platan** [comment]
Thanks for a review and suggestions. I've applied all of them.

**Comment by gdzien** [comment]
Shouldn't we update README for that?

**Comment by glauberferreira** [comment]
Is there any roadmap to release a new version of ansible-role-nginx, including this PR? `auth_basic` and `auth_basic_file` are described at current public README, but they does not work since this PR has not been released yet.

**Comment by platan** [comment]
@glauberferreira As an alternative, you can always use this repo in `requirements.yml` ([documentation](https://docs.ansible.com/ansible/latest/reference_appendices/galaxy.html#installing-multiple-roles-from-a-file), I do it in [my project](https://github.com/platan/metrics-shields-io-config/blob/master/requirements.yml#L8-L10)).

**Comment by alessfg** [comment]
Hi @glauberferreira! I've noted your comment and we'll try to release a new version as soon as it's possible. For now, as @platan mentioned, please use this repo as the role source instead of Ansible Galaxy 😄

**Pull request merged**